db/schema_tables.cc

```diff
@@ -3651,7 +3651,7 @@ view_ptr maybe_fix_legacy_secondary_index_mv_schema(replica::database& db, const
     if (v->clustering_key_size() == 0) {
         return view_ptr(nullptr);
     }
-    const column_definition& first_view_ck = v->clustering_key_columns().front();
+    const column_definition first_view_ck = v->clustering_key_columns().front();
```

Comment:
Here too, since the `view_ptr` is guaranteed to be valid by caller, I think it would make more sense to do:
```c++
    const auto ck_cols = v->clustering_key_columns();
    const column_definition& first_view_ck = ck_cols.front();
```

Reply:
indeed. thanks for your review and suggestion. incorporated.